src/zip.js

```diff
@@ -110,19 +117,180 @@ function zipJs(functionPath, zipPath) {
+                entryPath: updatedEntry
+              }
+              // newPath = removeQuotes(requirePath)
+            }
+
+            /* if relvative path clashes, we need to shift & update file names */
+            if (processedPaths.includes(newPath)) {
+              const end = newPath.replace(/['|"]$/g, '')
+              const postFix = prettyPath(modulePath, moduledir)
+              const updatedPath = (end.match(/^'/)) ? `${end}-${postFix}'` : `${end}-${postFix}"`
+              console.log(`Duplicate path detected at "${newPath}". Shifting path to ${updatedPath}`)
+              // Map new entry point for file
+              mappings[modulePath] = {
+                file: file,
+                entryPath: formatEntryPoint(updatedPath)
+              }
+              newPath = updatedPath
+            }
+
+            processedPaths.push(newPath)
+
+            const fixedRelativePath = `require(${newPath})`
+            // console.log('fixedPath', fixedRelativePath)
+            const replaceRegex = new RegExp(`require\\(${requirePath}\\)`, 'gm')
+            // console.log('replaceRegex', replaceRegex)
+            content = content.replace(replaceRegex, fixedRelativePath)
+          }
+          // console.log('processedPaths', processedPaths)
+        })
+      }
+
+      const stat = fs.lstatSync(file);
+      zip.addLocalFile(Buffer.from(content), {
+        name: entryPath,
+        mode: stat.mode,
+        date: new Date(0), // Ensure sha256 stability regardless of mtime
+        stats: stat
+      });
+    }
   });
   return zip.finalize();
 }
 
+function formatEntryPoint(reqPath) {
+  const entry = reqPath
+    // remove leading ./
+    .replace(/^\.\//, '')
+    // Remove file extension
+    .replace(/\.js$/, '')
+  return `${removeQuotes(entry)}.js`
+}
+
+function inNodeModules(filePath) {
+  return filePath.match(/\/node_modules\//)
+}
+
+function removeFileExtension(str) {
+  return str.replace(/\.js$/, '')
+}
+
+function isPathInside(childPath, parentPath) {
+	childPath = path.resolve(childPath);
+	parentPath = path.resolve(parentPath);
+
+	if (process.platform === 'win32') {
+		childPath = childPath.toLowerCase();
+		parentPath = parentPath.toLowerCase();
+	}
+
+	if (childPath === parentPath) {
+		return false;
+	}
+
+	childPath += path.sep;
+	parentPath += path.sep;
+
+	return childPath.startsWith(parentPath);
+}
+
+function prettyPath(filePath, moduledir) {
+  return filePath
+    // Remove base dir from path
+    .replace(moduledir, '')
+    // remove leading slash
+    .replace(/^\//, '')
+    // replace slashes with dashes
+    .replace(/\//g, '-')
+    // replace trailing file type
+    .replace(/\.js$/, '')
+}
+
+function removeQuotes(str) {
+  return str.replace(/['|"]/g, '')
+}
+
+function getOriginalPath(requirePath, file) {
+  const cleanp = removeQuotes(requirePath)
+  // Path ./whatever
+  if (cleanp.match(/^\./)) {
+    return require.resolve(path.join(path.dirname(file), cleanp))
```

Comment:
Using the `resolve` node module (that we use in the `finders.js`) might be safer?